### DISCUSSION TITLE
Best way to add a shadow plane?

### DISCUSSION THREAD

**Discussion by georgy95** [mitsuba-renderer/mitsuba3#847] [Q&A] [answered]
Given there isn't a shadow catcher material, is there an approximation of it using existing materials? 

What is the best way to achieve the effects of it?

**Comment by njroussel** [comment, marked as answer]
Hi @georgy95 

I don't believe there is a way to do this out-of-the box.
My suggestion would be to write a custom integrator and BSDF for this.

**Reply by georgy95** [reply]
Thanks!

**Comment by StarsTesla** [comment]
@georgy95 Hi, have you wrote the shadow catcher successfully?
